src/sentry/models/avatars/base.py

```diff
@@ -35,17 +38,31 @@ def save(self, *args, **kwargs):
         return super().save(*args, **kwargs)
 
     def get_file(self):
-        from sentry.models import File
-
-        if self.file_id is None:
-            return None
+        # Favor control_file_id if it exists and is set.
+        # Otherwise fallback to file_id. If still None, return.
+        file_class = self.file_class()
+        file_id = getattr(self, self.file_fk())
+        if file_id is None:
+            file_id = self.file_id
+            file_class = File
+            if file_id is None:
+                return None
+            copy_file_to_control_and_update_model.apply_async(
+                kwargs={
+                    "app_name": "sentry",
+                    "model_name": "UserAvatar",
```

Comment:
Is this always the case? This method is defined in a base class, so you could be handling any type of Avatar sub-class.

Reply:
Goood catch.